### PR TITLE
Fix for #3637 - use correct authority in multi-cloud + WAM scenario

### LINKED ISSUE #3637: [Bug] Incorrect authority in AuthResult using instance_aware
I used the below mentioned configuration and tried logging into a non-public cloud. I got logged in, successfully got access token as well, everything is correct Except Authority is not of respective cloud. It seems library is overriding the authority it gets in their result.

Also, when I enable instance_aware manually using query params it works fine except authority but when I use WithMultiCloudSupport(true) it doesn't work.

**Which version of MSAL.NET are you using?**
MSAL.NET 4.46.0

**Platform**
.NET Framework: 4.7

**What authentication flow has the issue?**
* Desktop
    * Integrated Windows Authentication

**Is this a new or existing app?**
The app is in production, I haven't upgraded MSAL, but started seeing this issue.

**Repro**
1.Use this configuration of WAM
```csharp
return PublicClientApplicationBuilder.Create(clientId)
.WithDefaultRedirectUri()
.WithAuthority(authority, tenant)
.WithExtraQueryParameters(new Dictionary<string, string>()
{{"instance_aware", "true" }
})
.WithWindowsBroker().WithMsalLogging();
```
2.Login using a non-public cloud account which is not added to your OS.

**Expected behavior**
Environment of respective cloud in AuthResult (account used is non-public).

**Actual behavior**
Got Public Environment in AuthResult but the account was non-public cloud.

**Possible solution**
Do not override the auth result, instead return the authority as it is at least for non-public cloud OR fix WithMultiCloudSupport(true) with WAM.

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #3637 

**Changes proposed in this request**

**Testing**
unit + manual

**Performance impact**
there is new logic in hot path, but it is behind the "IsMultiCloud" flag.

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <TargetFrameworkNet5Win>net5.0-windows10.0.17763.0</TargetFrameworkNet5Win>
     <TargetFrameworkNetStandard>netstandard2.0</TargetFrameworkNetStandard>
     <Configurations>Debug;Release;Debug + MobileApps</Configurations>
+
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj:11]: nit : extra line

File: src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs [hunk 2]
```diff
@@ -157,7 +157,7 @@ private enum ResponseStatus
 
             if (authenticationRequestParameters.AppConfig.MultiCloudSupportEnabled)
             {
-                authParams.Properties["discover"] = "home";
+                authParams.Properties["instance_aware"] = "true";
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs:160]: Do we still have ["discover"] = "home", if so what is that used for?

**Review by SameerK-MSFT — APPROVED** [review]

File: src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs (hunk 2 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs:160]: "discover" = "home" is the magic sauce if you talk directly to WAM. 
"instance_aware" = "true" is the magic sauce if you talk to MSAL C++ or directly to ESTS. MSAL C++ will transform "instace_aware"="true" into "discover"="home" when talking to WAM on its own.

File: tests/Microsoft.Identity.Test.Unit/BrokerTests/BrokerRequestTests.cs [hunk 3]
```diff
@@ -856,6 +846,41 @@ public void SilentStrategy_ProtectionPolicyNotEnabled_Throws_Exception()
...
+                var tokenResponse = CreateTokenResponseForTest();
+                tokenResponse.AuthorityUrl = "https://login.microsoftonline.us/organizations";
+                broker.AcquireTokenInteractiveAsync(null, null).ReturnsForAnyArgs(Task.FromResult(tokenResponse));
+
+                // Act - interactive flow logs-in Arlighton user
+                var result = await globalPca.AcquireTokenInteractive(TestConstants.s_graphScopes).ExecuteAsync().ConfigureAwait(false);
+                Assert.AreEqual("login.microsoftonline.us", result.Account.Environment);
+                
+                var account = (await globalPca.GetAccountsAsync().ConfigureAwait(false)).Single();
+                Assert.AreEqual("login.microsoftonline.us", account.Environment);
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/BrokerTests/BrokerRequestTests.cs:882]:
```suggestion
                Assert.AreEqual("login.microsoftonline.us", result.Account.Environment);
                Assert.AreEqual(TestConstants.Organizations, result.TenantId);
                
                var account = (await globalPca.GetAccountsAsync().ConfigureAwait(false)).Single();
                Assert.AreEqual("login.microsoftonline.us", account.Environment);
                Assert.AreEqual(TestConstants.Organizations, result.TenantId);
```

**Review by pmaytak — CHANGES_REQUESTED** [review]

**Review by pmaytak — APPROVED** [review]

**Pull request merged**